### PR TITLE
feat(localization): Core localization infrastructure with reactive culture switching

### PR DESCRIPTION
**Pull request by OmarAglan**
This PR implements Core localization infrastructure of the comprehensive localization system for GenHub, establishing the foundational infrastructure for multi-language support. This phase introduces the core services, interfaces, and architecture that will enable seamless translation of the application into multiple languages with runtime language switching.

---

## Core Services & Interfaces

- **`ILocalizationService`** - Primary service interface for accessing localized strings and managing culture
- **`LocalizationService`** - Singleton service implementation with reactive culture change notifications
- **`ILanguageProvider`** - Interface for language discovery and resource management
- **`LanguageProvider`** - Implementation that discovers available languages via satellite assemblies
- **`LocalizationOptions`** - Configuration model for localization behavior
- **`LocalizationExtensions`** - Helper methods for common localization scenarios

## Key Features

**Reactive Culture Switching** - Uses `System.Reactive` for UI-friendly culture change notifications  
**Thread-Safe Operations** - All culture switching and resource access is thread-safe  
**Graceful Fallback** - Automatically falls back to English for missing translations  
**Dynamic Language Discovery** - Scans for available languages from satellite assemblies  
**Culture-Aware Formatting** - Extensions for dates, numbers, currency, file sizes, etc.  
**Comprehensive Testing** - 44 unit tests with 100% pass rate  

### Service Layer

```
┌─────────────────────────────────────────┐
│         ILocalizationService            │
│  - CurrentCulture: CultureInfo         │
│  - AvailableCultures: List<Culture>    │
│  - CultureChanged: IObservable         │
│  - GetString(key): string              │
│  - SetCulture(culture): Task           │
└─────────────────┬───────────────────────┘
                  │
                  │ depends on
                  ▼
┌─────────────────────────────────────────┐
│         ILanguageProvider               │
│  - DiscoverAvailableLanguages()        │
│  - GetResourceManager(baseName)        │
│  - ValidateCulture(culture)            │
└─────────────────────────────────────────┘
```

### Reactive Pattern

The service implements reactive culture switching using `System.Reactive`:

```csharp
// In ViewModels, subscribe to culture changes
_localization.CultureChanged
    .ObserveOn(RxApp.MainThreadScheduler)
    .Subscribe(_ => UpdateLocalizedStrings());
```

This ensures all UI elements update automatically when the user changes languages.

### Thread Safety

- Uses `lock` statements for culture switching operations
- Thread-safe caching of `ResourceManager` instances
- Proper synchronization prevents race conditions during concurrent access

---

## Test Coverage

- **28 LocalizationService tests** covering:
  - Culture switching (valid/invalid cultures)
  - String retrieval with/without parameters
  - Fallback behavior
  - Reactive notifications
  - Thread safety
  - Configuration options
  - Disposal pattern

- **16 LanguageProvider tests** covering:
  - Language discovery
  - Culture validation
  - ResourceManager provisioning
  - Caching behavior
  - Error handling

## Basic Setup (Dependency Injection)

```csharp
// In App.axaml.cs or startup configuration
services.AddLocalizationServices(options =>
{
    options.DefaultCulture = "en";
    options.FallbackCulture = "en";
    options.LogMissingTranslations = true;
});
```

## In ViewModels

```csharp
public class MyViewModel : ViewModelBase
{
    private readonly ILocalizationService _localization;

    public MyViewModel(ILocalizationService localization)
    {
        _localization = localization;
        
        // Subscribe to culture changes for reactive updates
        _localization.CultureChanged
            .ObserveOn(RxApp.MainThreadScheduler)
            .Subscribe(_ => this.RaisePropertyChanged(nameof(WelcomeText)));
    }

    // Property that automatically updates when culture changes
    public string WelcomeText => _localization.GetString("UI.Common.Welcome");
    
    // Formatted string with parameters
    public string GreetingText => _localization.GetString(
        "UI.Common.Greeting", 
        _userService.CurrentUser.Name
    );
}
```

## Changing Languages at Runtime

```csharp
// Switch to German
await _localization.SetCulture("de");

// Or by CultureInfo
await _localization.SetCulture(new CultureInfo("fr-FR"));

// Get available languages for UI picker
var languages = _localization.AvailableCultures;
```

## Using Extension Methods

```csharp
// Culture-aware formatting
var formattedDate = _localization.FormatDate(DateTime.Now);
var formattedCurrency = _localization.FormatCurrency(1234.56m);
var formattedFileSize = _localization.FormatFileSize(1024 * 1024 * 500); // "500 MB"

// Try get with fallback
var success = _localization.TryGetString("UI.Common.Save", out var saveText);

// Pluralization support
var itemCount = _localization.GetPluralString(
    count: 5,
    noneKey: "UI.Common.NoItems",
    oneKey: "UI.Common.OneItem", 
    manyKey: "UI.Common.ManyItems"
);
```
- Implements #24

<h3>Greptile Summary</h3>

Implements comprehensive localization infrastructure using .NET resource files with reactive culture switching and extensive helper utilities. The architecture is well-designed with clean separation of concerns, strong typing via `StringResources` constants, and excellent test coverage (44 tests, 100% pass rate).

## Key Changes

- **Core Services**: `LocalizationService` and `LanguageProvider` with reactive `IObservable<CultureInfo>` for UI updates
- **Resource Management**: 11 `.resx` files organized by domain (UI.Common, UI.Settings, Errors, Messages, Tooltips)
- **Helper Extensions**: Culture-aware formatting for dates, numbers, currency, file sizes, and pluralization
- **DI Configuration**: Singleton registration with flexible configuration options
- **Documentation**: 1800+ line architecture design doc and resource guide

## Critical Issues Found

- **Performance**: `LanguageProvider` iterates through 800+ system cultures on startup instead of scanning satellite assembly directories
- **Deadlock Risk**: Property setters use `GetAwaiter().GetResult()` which will deadlock in UI contexts
- **Thread Pool Abuse**: `SetCulture()` unnecessarily wraps synchronous operations in `Task.Run()`

## Strengths

- Excellent architecture with proper abstraction layers
- Comprehensive extension methods for culture-aware formatting  
- Thread-safe with proper locking patterns
- Strong test coverage across all components
- Well-documented with inline comments and external docs
- Graceful fallback handling for missing translations

<h3>Confidence Score: 3/5</h3>

- Safe to merge with fixes - the architecture is solid but contains performance issues and deadlock risks that should be addressed before production use
- The localization system has excellent architecture, comprehensive testing, and good documentation. However, the language discovery mechanism will cause noticeable UI freezes (iterating 800+ cultures), and the blocking async calls in property getters/setters create real deadlock risks in Avalonia UI contexts. These are not edge cases - they will affect normal application startup and language switching operations.
- Pay close attention to `LanguageProvider.cs` (performance issue) and `LocalizationService.cs` (deadlock risks in property implementations)

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| GenHub/GenHub.Core/Services/Localization/LocalizationService.cs | Core service implementing reactive culture switching with thread-safe operations. Contains blocking async call in property setter that could cause deadlock in UI contexts. |
| GenHub/GenHub.Core/Services/Localization/LanguageProvider.cs | Discovers satellite assemblies by iterating ALL system cultures (~800+), causing severe performance degradation on first access. Should use directory scanning instead. |
| GenHub/GenHub.Core/Interfaces/Localization/ILocalizationService.cs | Well-designed interface with clear separation of concerns, reactive pattern support, and comprehensive documentation. |
| GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs | Comprehensive helper methods for culture-aware formatting. Well-tested and properly documented. |
| GenHub/GenHub/Infrastructure/DependencyInjection/LocalizationServicesModule.cs | Clean DI registration with singleton lifetimes and configuration overloads. |
| GenHub/GenHub.Tests/GenHub.Tests.Core/Services/Localization/LocalizationServiceTests.cs | Comprehensive test coverage with 28 tests covering edge cases, thread safety, and disposal patterns. |

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant UI as UI Component/ViewModel
    participant LS as LocalizationService
    participant LP as LanguageProvider
    participant RM as ResourceManager
    participant SA as Satellite Assemblies
    participant Obs as Observers (UI Subscribers)
    
    Note over UI,SA: Application Startup
    UI->>LS: new LocalizationService(provider, logger, options)
    LS->>LS: Initialize with DefaultCulture ("en")
    LS->>LS: SetThreadCulture(culture)
    LS->>LS: Set CultureInfo.CurrentCulture
    LS->>LS: Set Thread.CurrentThread.CurrentCulture
    
    Note over UI,SA: User Requests Language Change
    UI->>LS: SetCulture("de")
    LS->>LP: ValidateCulture(culture)
    LP->>SA: assembly.GetSatelliteAssembly(culture)
    SA-->>LP: satellite assembly or FileNotFoundException
    LP-->>LS: true/false
    
    alt Culture is valid
        LS->>LS: lock(_lock)
        LS->>LS: _currentCulture = culture
        LS->>LS: SetThreadCulture(culture)
        LS->>Obs: _cultureChangedSubject.OnNext(culture)
        Obs->>UI: Culture changed notification
        UI->>UI: Update UI bindings (RaisePropertyChanged)
    else Culture is invalid
        LS->>LS: Fall back to FallbackCulture
        LS->>LS: SetThreadCulture(fallbackCulture)
        LS->>Obs: _cultureChangedSubject.OnNext(fallbackCulture)
    end
    
    Note over UI,SA: Retrieving Localized String
    UI->>LS: GetString(resourceSet, key)
    LS->>LP: GetResourceManager(baseName)
    
    alt ResourceManager cached
        LP-->>LS: cached ResourceManager
    else ResourceManager not cached
        LP->>RM: new ResourceManager(baseName, assembly)
        LP->>LP: Cache ResourceManager
        LP-->>LS: new ResourceManager
    end
    
    LS->>LS: lock(_lock)
    LS->>RM: GetString(key, _currentCulture)
    RM->>SA: Load from satellite assembly for culture
    SA-->>RM: localized string
    RM-->>LS: string value
    
    alt String found
        LS-->>UI: localized string
    else String not found
        LS->>LS: HandleMissingTranslation(key)
        LS->>LS: Log warning if enabled
        LS-->>UI: fallback string [key] or key
    end
```

### CODE REVIEW

**Comment by OmarAglan** [comment]
implemenled **Phase 2** of the comprehensive localization system for GenHub, building on the infrastructure established in Phase 1. This phase creates the complete `.resx` resource file structure with 278 English strings extracted from the existing codebase, configures satellite assembly generation, and validates the end-to-end resource loading workflow.


## What's New

### Resource File Structure (10 files, 278 strings)

Created organized `.resx` resource files with English translations:

- **UI.Common.resx** (32 strings) - Buttons, states, common actions
- **UI.Navigation.resx** (7 strings) - Tab names and navigation labels
- **UI.GameProfiles.resx** (25 strings) - Game profile management UI
- **UI.Settings.resx** (38 strings) - Settings page labels and options
- **UI.Updates.resx** (25 strings) - Update functionality messages
- **Errors.Validation.resx** (20 strings) - Input validation errors
- **Errors.Operations.resx** (38 strings) - Operation error messages
- **Messages.Success.resx** (27 strings) - Success confirmations
- **Messages.Confirmations.resx** (28 strings) - User confirmation dialogs
- **Tooltips.resx** (38 strings) - UI element tooltips

### Key Features

✅ **Hierarchical Organization** - Resources organized by functional area  
✅ **Translator-Friendly** - All strings have contextual comments  
✅ **Format String Support** - Parameterized strings with `{0}`, `{1}` placeholders  
✅ **Strongly-Typed Access** - `StringResources` class for compile-time safety  
✅ **Satellite Assembly Ready** - Configured for culture-specific translations  
✅ **Full Test Coverage** - 15 integration tests validating resource loading  
✅ **Comprehensive Documentation** - Complete guide for developers and translators

---

## Files Changed

### New Files (13 files, 3,428 lines)

**Resource Files** (10 files, 2,780 lines)
- `GenHub.Core/Resources/Strings/UI.Common.resx` (32 strings)
- `GenHub.Core/Resources/Strings/UI.Navigation.resx` (7 strings)
- `GenHub.Core/Resources/Strings/UI.GameProfiles.resx` (25 strings)
- `GenHub.Core/Resources/Strings/UI.Settings.resx` (38 strings)
- `GenHub.Core/Resources/Strings/UI.Updates.resx` (25 strings)
- `GenHub.Core/Resources/Strings/Errors.Validation.resx` (20 strings)
- `GenHub.Core/Resources/Strings/Errors.Operations.resx` (38 strings)
- `GenHub.Core/Resources/Strings/Messages.Success.resx` (27 strings)
- `GenHub.Core/Resources/Strings/Messages.Confirmations.resx` (28 strings)
- `GenHub.Core/Resources/Strings/Tooltips.resx` (38 strings)

**Helper Classes** (1 file, 64 lines)
- `GenHub.Core/Resources/Strings/StringResources.cs` - Strongly-typed resource namespace constants

**Tests** (1 file, 384 lines)
- `GenHub.Tests/GenHub.Tests.Core/Resources/StringResourcesTests.cs` - Integration tests for resource loading

**Documentation** (1 file, 200 lines)
- `docs/localization-resources-guide.md` - Complete resource usage guide

### Modified Files (4 files)

**Project Configuration**
- `GenHub.Core/GenHub.Core.csproj` - Added resource generation and satellite assembly configuration

**Service Updates**
- `GenHub.Core/Interfaces/Localization/ILocalizationService.cs` - Added `GetString(resourceSet, key)` overloads
- `GenHub.Core/Services/Localization/LocalizationService.cs` - Implemented resource set support

**Test Updates**
- `GenHub.Tests/GenHub.Tests.Core/Services/Localization/LocalizationServiceTests.cs` - Added tests for new overloads

---

## Architecture & Design

### Resource Organization

```
GenHub.Core/Resources/Strings/
├── UI.Common.resx              → GenHub.Core.Resources.Strings.UI.Common
├── UI.Navigation.resx          → GenHub.Core.Resources.Strings.UI.Navigation
├── UI.GameProfiles.resx        → GenHub.Core.Resources.Strings.UI.GameProfiles
├── UI.Settings.resx            → GenHub.Core.Resources.Strings.UI.Settings
├── UI.Updates.resx             → GenHub.Core.Resources.Strings.UI.Updates
├── Errors.Validation.resx      → GenHub.Core.Resources.Strings.Errors.Validation
├── Errors.Operations.resx      → GenHub.Core.Resources.Strings.Errors.Operations
├── Messages.Success.resx       → GenHub.Core.Resources.Strings.Messages.Success
├── Messages.Confirmations.resx → GenHub.Core.Resources.Strings.Messages.Confirmations
└── Tooltips.resx               → GenHub.Core.Resources.Strings.Tooltips
```

### Resource Key Naming Convention

Uses hierarchical dot notation for clarity:

```
UI.Common.Button.Save           → "Save"
UI.Common.Button.Cancel         → "Cancel"
UI.GameProfiles.Create.Title    → "Create New Profile"
Errors.Validation.RequiredField → "{0} is required."
Messages.Confirmations.Delete   → "Are you sure you want to delete {0}?"
```

### Project Configuration

Added to `GenHub.Core.csproj`:

```xml
<PropertyGroup>
  <GenerateSatelliteAssembliesForCore>true</GenerateSatelliteAssembliesForCore>
</PropertyGroup>

<ItemGroup>
  <EmbeddedResource Update="Resources\Strings\*.resx">
    <Generator>ResXFileCodeGenerator</Generator>
    <CustomToolNamespace>GenHub.Core.Resources.Strings</CustomToolNamespace>
  </EmbeddedResource>
</ItemGroup>
```

This configures:
- Satellite assembly generation for culture-specific resources
- Automatic code generation for strongly-typed resource access
- Custom namespace for generated resource classes

---

##  Testing

### Test Coverage

Added **15 integration tests**, all passing ✅

Tests validate:
- ✅ All 10 resource sets are properly embedded
- ✅ ResourceManager creation for each namespace
- ✅ String retrieval from each resource set
- ✅ LocalizationService integration with resource sets
- ✅ Format string parameter substitution
- ✅ ResourceManager caching
- ✅ Multiple strings per resource set
- ✅ Resource set constant correctness

## 📖 Usage Examples

### Basic String Retrieval

```csharp
public class MyViewModel : ViewModelBase
{
    private readonly ILocalizationService _localization;

    public MyViewModel(ILocalizationService localization)
    {
        _localization = localization;
    }

    public void SaveProfile()
    {
        // Get a simple string
        var saveText = _localization.GetString(
            StringResources.UiCommon, 
            "Button.Save"
        );
        // Returns: "Save"
    }
}
```

### Format Strings with Parameters

```csharp
// String in Errors.Validation.resx: "RequiredField" = "{0} is required."
var errorMessage = _localization.GetString(
    StringResources.ErrorsValidation,
    "RequiredField",
    "Profile Name"
);
// Returns: "Profile Name is required."
```

### Using StringResources Constants

```csharp
// Strongly-typed resource set names prevent typos
var confirmDelete = _localization.GetString(
    StringResources.MessagesConfirmations,  // Compile-time checked!
    "DeleteProfile",
    profileName
);
// Returns: "Are you sure you want to delete {profileName}?"
```

### Real-World ViewModel Example

```csharp
public class SettingsViewModel : ViewModelBase
{
    private readonly ILocalizationService _localization;

    public SettingsViewModel(ILocalizationService localization)
    {
        _localization = localization;
        
        // Subscribe to culture changes for reactive updates
        _localization.CultureChanged
            .ObserveOn(RxApp.MainThreadScheduler)
            .Subscribe(_ => UpdateLocalizedProperties());
    }

    // Property bound to UI
    public string ThemeLabel => _localization.GetString(
        StringResources.UiSettings,
        "Theme.Label"
    );
    
    // Property bound to UI
    public string LanguageLabel => _localization.GetString(
        StringResources.UiSettings,
        "Language.Label"
    );

    private void UpdateLocalizedProperties()
    {
        // Notify UI that all localized properties have changed
        this.RaisePropertyChanged(nameof(ThemeLabel));
        this.RaisePropertyChanged(nameof(LanguageLabel));
        // ... more properties
    }
}
```

---

## 📚 Resource File Structure

### Sample: UI.Common.resx

```xml
<data name="Button.Save" xml:space="preserve">
  <value>Save</value>
  <comment>Button text for saving changes</comment>
</data>

<data name="Status.Loading" xml:space="preserve">
  <value>Loading...</value>
  <comment>Displayed while content is loading</comment>
</data>

<data name="Action.Delete" xml:space="preserve">
  <value>Delete</value>
  <comment>Action to delete an item</comment>
</data>
```

### Sample: Errors.Validation.resx

```xml
<data name="RequiredField" xml:space="preserve">
  <value>{0} is required.</value>
  <comment>Generic validation error for required fields. {0} = field name</comment>
</data>

<data name="InvalidPath" xml:space="preserve">
  <value>The path '{0}' is not valid.</value>
  <comment>Path validation error. {0} = the invalid path</comment>
</data>

<data name="ProfileNameTooLong" xml:space="preserve">
  <value>Profile name must be {0} characters or less.</value>
  <comment>Profile name length validation. {0} = maximum length</comment>
</data>
```

---

## 🔍 String Extraction Coverage

Strings were systematically extracted from:

### ViewModels ✅
- `MainViewModel` - Navigation, profile creation, status messages
- `SettingsViewModel` - All settings labels, theme options, validation
- `GameProfileLauncherViewModel` - Profile loading, game scanning messages
- `UpdateNotificationViewModel` - Update checks, download progress, installation
- `ContentBrowserViewModel` - Search, filter labels

### Services ✅
- `LocalizationService` - Error messages, culture validation
- `LanguageProvider` - Discovery errors, logging messages

### Constants ✅
- `UiConstants` - UI labels and messages
- `ValidationLimits` - Validation error templates

### What Was NOT Extracted

The following remain in English as per best practices:
- ❌ Debug/trace log messages
- ❌ Developer exception messages
- ❌ Technical identifiers and enum names
- ❌ Code comments and documentation

---

## Translator-Friendly Features

### 1. Contextual Comments
Every string includes a comment explaining:
- Where it's used in the UI
- What parameters represent (if any)
- Character length constraints (if applicable)

### 2. Logical Grouping
Related strings are grouped together:
```
Button.Save
Button.Cancel
Button.OK
Button.Apply
```

### 3. Clear Naming
Key names are self-documenting:
- `UI.GameProfiles.Create.Title` clearly indicates a title for profile creation
- `Errors.Validation.RequiredField` is obviously a validation error

### 4. Format Placeholders
Numbered placeholders are documented:
```xml
<comment>
  {0} = profile name
  {1} = game name
  Example: "Profile 'MyProfile' created for The Sims 4"
</comment>
```

### 5. Translation Tool Support
`.resx` files work with:
- **ResXManager** (recommended, free, cross-platform)
- **Visual Studio Resource Editor**
- **Rider Resource Editor**
- Any text editor (XML format)

---

## 🚀 Satellite Assembly Generation

### How It Works

When culture-specific translations are added (Phase 8-9):

```
GenHub.Core/Resources/Strings/
├── UI.Common.resx              → Embedded in GenHub.Core.dll (English)
├── UI.Common.de.resx           → German satellite: de/GenHub.Core.resources.dll
├── UI.Common.fr.resx           → French satellite: fr/GenHub.Core.resources.dll
└── UI.Common.es.resx           → Spanish satellite: es/GenHub.Core.resources.dll
```

Build output structure:
```
bin/Debug/net8.0/
├── GenHub.Core.dll              (English resources embedded)
├── de/
│   └── GenHub.Core.resources.dll (German satellite)
├── fr/
│   └── GenHub.Core.resources.dll (French satellite)
└── es/
    └── GenHub.Core.resources.dll (Spanish satellite)
```

### Current Status

- ✅ Configuration complete
- ✅ Infrastructure ready
- ⏳ Satellite assemblies will be generated when translations are added

---

## Metrics

| Metric | Value |
|--------|-------|
| Resource Files | 10 |
| Total Strings | 278 |
| Lines of Resources | 2,780 |
| Test Coverage | 15 tests, 100% pass |
| Resource Categories | UI (4), Errors (2), Messages (2), Tooltips (1), Navigation (1) |
| Average Strings per File | 27.8 |

---

## Additional Resources Or Docs

- **Architecture Design**: [`docs/architecture/localization-system-design.md`](docs/architecture/localization-system-design.md)
- **Resource Guide**: [`docs/localization-resources-guide.md`](docs/localization-resources-guide.md) (NEW)
- **Avalonia Localization Docs**: https://docs.avaloniaui.net/docs/guides/implementation-guides/localizing

---

## Translation Workflow (For Future)

When adding new languages:

1. **Copy English .resx files** and rename with culture code:
   - `UI.Common.resx` → `UI.Common.de.resx` (German)
   - `UI.Common.resx` → `UI.Common.fr.resx` (French)

2. **Translate values** (keep keys unchanged):
   ```xml
   <!-- UI.Common.de.resx -->
   <data name="Button.Save" xml:space="preserve">
     <value>Speichern</value>
   </data>
   ```

3. **Build project** - Satellite assemblies auto-generate

4. **Test** - Language appears in settings automatically

---

File: docs/architecture/localization-system-design.md [hunk 1]
```diff
@@ -0,0 +1,1872 @@
...
+1. **Requested Culture** (e.g., `de-DE`)
+2. **Neutral Culture** (e.g., `de`)
+3. **Neutral Resource** (e.g., English from `UI.Common.resx`)
+
+**Example**:
+
+- User selects German (`de-DE`)
+- ResourceManager looks for `UI.Common.de-DE.resx` (doesn't exist)
+- Falls back to `UI.Common.de.resx` (exists, uses this)
+- If `de.resx` missing a key, falls back to `UI.Common.resx` (English)
```
Comment (undead2146) [inline docs/architecture/localization-system-design.md]:
```suggestion
- If `de.resx` is missing a key, falls back to `UI.Common.resx` (English)
```

File: docs/architecture/localization-system-design.md [hunk 2]
```diff
@@ -0,0 +1,1872 @@
...
+**Next Steps**:
+
+1. Review and approve this architecture
+2. Create GitHub issues for each phase
+3. Begin Phase 1 implementation
+4. Install ResXManager and familiarize team
+
+---
+
+**Document End**
```
Comment (undead2146) [inline docs/architecture/localization-system-design.md]: redundant

File: GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+        string key,
+        out string? value)
+    {
+        ArgumentNullException.ThrowIfNull(service, nameof(service));
+        ArgumentException.ThrowIfNullOrWhiteSpace(key, nameof(key));
+
+        try
+        {
+            var result = service.GetString(key);
+
```
Comment (undead2146) [inline GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs]: Code should not contain trailing whitespace

File: GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+    public static string FormatDateTime(
+        this ILocalizationService service,
+        DateTime dateTime,
+        string? format = null)
+    {
+        ArgumentNullException.ThrowIfNull(service, nameof(service));
+
+        var culture = service.CurrentCulture;
+        return string.IsNullOrWhiteSpace(format)
+            ? dateTime.ToString("g", culture)  // Short date + time pattern
```
Comment (undead2146) [inline GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs]: Code should not contain multiple whitespace characters in a row.

File: GenHub/GenHub.Core/Services/Localization/LanguageProvider.cs [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                        "Discovered {Count} available languages: {Languages}",
+                        _cachedCultures.Count,
+                        string.Join(", ", _cachedCultures.Select(c => c.Name)));
+
+                    return _cachedCultures;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error discovering available languages");
+
```
Comment (undead2146) [inline GenHub/GenHub.Core/Services/Localization/LanguageProvider.cs]: Code should not contain trailing whitespace

File: GenHub/GenHub.Core/Services/Localization/LanguageProvider.cs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+            if (_resourceManagers.TryGetValue(baseName, out var existingManager))
+            {
+                return existingManager;
+            }
+
+            try
+            {
+                var assembly = Assembly.GetExecutingAssembly();
+                var resourceManager = new ResourceManager(baseName, assembly);
+
```
Comment (undead2146) [inline GenHub/GenHub.Core/Services/Localization/LanguageProvider.cs]: Code should not contain trailing whitespace

File: GenHub/GenHub.Core/Services/Localization/LocalizationService.cs [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+        }
+        catch (FormatException ex)
+        {
+            _logger.LogError(
+                ex,
+                "Format error for key '{Key}' from resource set '{ResourceSet}' with {ArgCount} arguments",
+                key,
+                resourceSet,
+                args?.Length ?? 0);
+
```
Comment (undead2146) [inline GenHub/GenHub.Core/Services/Localization/LocalizationService.cs]: Code should not contain trailing whitespace

**Review by undead2146 — CHANGES_REQUESTED** [review]
Looks good so far, I don't have much experience with localisation hence I cannot give a deeper review than just the surface level comments

**Comment by undead2146** [comment]
Is this still being worked on @OmarAglan?

**Comment by OmarAglan** [comment]
Added 159 missing user-facing strings across 12 resource files

comprehensive string extraction audit for GenHub's localization system. The initial resource file had 278 strings, i scanned the entire codebase to identify and add all remaining user-facing strings, bringing the total to **437 localized strings** across **12 resource files**.

### New Resource Files

1. **UI.Tools.resx** (32 strings)
   - Tool plugin management UI
   - Tool installation/uninstallation messages
   - Tool settings and configuration
   - Tool validation messages

2. **UI.Downloads.resx** (18 strings)
   - Download status messages
   - Download progress labels
   - Download error messages
   - Queue management UI

**Comment by undead2146** [comment]
@OmarAglan is this PR ready for review?

File: GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs (hunk 3 above)
Comment (OmarAglan) [reply GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs]: Can You Explain what you mean be that?

File: GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+    public static string FormatDate(
+        this ILocalizationService service,
+        DateTime date,
+        string? format = null)
+    {
+        ArgumentNullException.ThrowIfNull(service, nameof(service));
+
+        var culture = service.CurrentCulture;
+        return string.IsNullOrWhiteSpace(format)
+            ? date.ToString("d", culture)  // Short date pattern
```
[the remark this answers is not in the document]
Comment (OmarAglan) [reply GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs]: Same as Above

File: GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs (hunk 4 above)
Comment (OmarAglan) [reply GenHub/GenHub.Core/Extensions/Localization/LocalizationExtensions.cs]: Same as Above

**Comment by undead2146** [comment]
@greptile

**Review by RMWMStevens — COMMENTED** [review]
Well done on this PR. Glad to see you got this working! I left some comments in the hope I can help you improve this a little bit more. Let me know what you think.

**Comment by OmarAglan** [comment]
@undead2146  will pick it up soon, with the aim to make it perfect!
